GDAXSharp/Services/AbstractService.cs

```diff
@@ -53,13 +56,17 @@ public abstract class AbstractService
                 errorMessage = contentBody;
             }
 
-            throw new GDAXSharpHttpException(errorMessage)
-            {
-                StatusCode = httpResponseMessage.StatusCode,
-                RequestMessage = httpRequestMessage,
-                ResponseMessage = httpResponseMessage,
-                EndPoint = new EndPoint(httpMethod, uri, content)
-            };
+            var ex = new GDAXSharpHttpException(errorMessage)
+                            {
+                                StatusCode = httpResponseMessage.StatusCode,
+                                RequestMessage = httpRequestMessage,
+                                ResponseMessage = httpResponseMessage,
+                                EndPoint = new EndPoint(httpMethod, uri, content)
+                            };
+
+            Log.Error("REST request about to throw {@GDAXSharpHttpException}", ex);
```

Comment:
what do you mean by about to throw? wouldn't the error have thrown

Reply:
The error throws on the next line...

In the ideal world the logging would happen where the error is eventually caught. However, I've worked on systems where "the error is handled" and so not logged... often this sort of low level network error indicates errors outside the norm or unexpected future behavior when the external system changes further down the line.

So I thought it would be useful to ensure this Exception is logged prior to actually throwing the exception.